### PR TITLE
Add UPGRADE.md for v0.22

### PR DESCRIPTION
**Pull request by evnu**
This adds a new `UPGRADE.md` file, which can be used to explain how to migrate to new versions.

### CODE REVIEW

**Comment by evnu** [comment]
@scrogson @jeroenvisser101 Did I miss something which should also be explained?

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,13 +5,16 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+See `UPGRADE.md` for additional help when upgrading to newer versions.
```
Comment (jeroenvisser101) [inline CHANGELOG.md:8]:
```suggestion
See [`UPGRADE.md`](./UPGRADE.md) for additional help when upgrading to newer versions.
```

I think this should link to it :)

File: UPGRADE.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Upgrade
+
+This document here is intended to simplify upgrading to newer versions by extending the changelog.
+
+## 0.21 -> 0.22
+
+0.22 changes how to define NIFs. Users upgrading to 0.22 should to do two things:
+
+1. Replace `rustler::rustler_export_nifs!` with `rustler::init!`
```
Comment (jeroenvisser101) [inline UPGRADE.md:9]: Maybe also note the change in `atoms!` and maybe resource/struct/etc?

Comment (evnu) [reply UPGRADE.md]: I think `atoms!` and `resource` are quite well described in the changelog, but I added more information on `rustler::init!`, as well as how to migrate schedule flags and renames. What do you think?

(Also, thanks for reading this PR! :))

File: UPGRADE.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+NIFs signature resembles the Elixir function's signature:
+
+```rust
+#[rustler::nif]
+fn add(a: i64, b: i64) -> i64
+```
+
+Under the hood, this is implemented by the `rustler::nif` proc_macro. For the
+new form to work, the parameters' types need to implement `Decoder`, and the
+return type needs to implement `Decoder`.
```
Comment (scrogson) [inline UPGRADE.md:79]: Encoder

File: UPGRADE.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
+# Upgrade
+
+This document here is intended to simplify upgrading to newer versions by extending the changelog.
```
Comment (scrogson) [inline UPGRADE.md:3]: This document is intended...

File: UPGRADE.md (hunk 2 above)
Comment (jeroenvisser101) [reply UPGRADE.md]: I think it's good! I think keeping it all in one place will save some time for those looking to upgrade :) Great work 👍

**Review by scrogson — APPROVED** [review]

**Pull request merged**